dashboard/app/models/pd/teacher_progress_report.rb

```diff
@@ -0,0 +1,56 @@
+class Pd::TeacherProgressReport
+
+  def self.generate_report_for_user(user)
+    teachers =
+      if user.admin?
+        # All teachers in all workshops
+        Pd::Attendance.distinct_teachers
+      elsif user.district_contact?
+        # All teachers attending workshops, from districts for which I am a contact
+        Pd::Attendance.for_district(user.district_as_contact).distinct_teachers
+      elsif user.workshop_organizer?
+        # All teachers in workshops I organized
+        Pd::Workshop.in_state(Pd::Workshop::STATE_ENDED).organized_by(user).map do |workshop|
+          Pd::Attendance.for_workshop(workshop).distinct_teachers
+        end.flatten
+      else
+        []
```

Comment:
If you get here then something is wrong with the permissions model. You should redirect to forbidden